### PR TITLE
Remove assets that have been replaced with SVG versions

### PR DESCRIPTION
**Pull request by janfaracik**
**Split from https://github.com/jenkinsci/jenkins/pull/5777 to keep the first PR independently mergeable as this PR is rather large..**

### Proposed changelog entries

* Remove images from core that have been replaced by SVG images.

### Proposed upgrade guidelines

Jenkins has removed a large number of static assets. Ensure all your plugins are the most recent release before updating core, otherwise image may be missing. Check for new plugin versions immediately after the core update and install available updates.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by timja** [comment]
Pretty much all the symbols in the context menu are broken:

image

(only issue I could find)

**Comment by timja** [comment]
conflicted

**Comment by NotMyFault** [comment]
The icon at the script console is broken as well.

**Comment by timja** [comment]
ATH tests passed: https://github.com/jenkinsci/acceptance-test-harness/pull/714

Bom tests fail: https://github.com/jenkinsci/bom/pull/729/checks?check_run_id=4291442648

https://github.com/jenkinsci/scm-api-plugin/blob/master/src/test/java/jenkins/scm/impl/avatars/AvatarCacheTest.java#L119

(running full set of bom tests now with failFast set to false to see if there's any more)

**Comment by timja** [comment]
It fails because the test is using search.png from core:
https://github.com/jenkinsci/scm-api-plugin/blob/master/src/test/java/jenkins/scm/impl/avatars/AvatarCacheTest.java#L129

The avatar cache only supports png so either need to switch to one not being removed or move an image to the plugin (probably better)

**Comment by timja** [comment]
Sorted in https://github.com/jenkinsci/scm-api-plugin/pull/102 by copying resource to plugin.

**Comment by janfaracik** [comment]
> Sorted in [jenkinsci/scm-api-plugin#102](https://github.com/jenkinsci/scm-api-plugin/pull/102) by copying resource to plugin.

That's amazing, thanks a bunch Tim :)

**Review by NotMyFault — COMMENTED** [review]
Are certain symbols kept for legacy purposes, for example, `go-next` is superseded by `next`, `document_edit` by `notepad`, `text` by `document` or `text-error` by `delete-document`, or does the removal only apply to icons that share the same name, not the same graphic?

File: core/src/main/resources/lib/hudson/executors.jelly
Comment (timja) [reply core/src/main/resources/lib/hudson/executors.jelly:42]: nice find:
image

Comment (janfaracik) [reply core/src/main/resources/lib/hudson/executors.jelly]: Good spot! Added :)

**Comment by janfaracik** [comment]
> Are certain symbols kept for legacy purposes, for example, `go-next` is superseded by `next`, `document_edit` by `notepad`, `text` by `document` or `text-error` by `delete-document`, or does the removal only apply to icons that share the same name, not the same graphic?

The removal only applies to icons that share the same name

**Review by timja — APPROVED** [review]
I've tested this quite a few times and all found issues have been resolved, thanks!

**Comment by res0nance** [comment]
If we need the additional PRs to be merged first does it make sense to put this on hold?

**Comment by timja** [comment]
> If we need the additional PRs to be merged first does it make sense to put this on hold?

those are additional PRs to improve the icons in the plugins, I'm not aware of any of those actually breaking this. but it means for the ones that copied icons to their plugins they'll pick up the core icon at some point.

If you can find an issue then sure we can put on hold

**Comment by NotMyFault** [comment]
> > If we need the additional PRs to be merged first does it make sense to put this on hold?
> 
> those are additional PRs to improve the icons in the plugins, I'm not aware of any of those actually breaking this. but it means for the ones that copied icons to their plugins they'll pick up the core icon at some point.
> 
> If you can find an issue then sure we can put on hold

PRs that contain changes like
```diff
- <l:task icon="images/24x24/up.png" />
+ <l:task icon="icon-up icon-md" />
```
should be merged and released before this PR is pulled, otherwise the plugin attempts to display icons from a path that is gone.

**Comment by timja** [comment]
Looking at open PRs / release status:

- https://github.com/jenkinsci/cloudbees-folder-plugin/pull/200 needs to be merged and released
- https://github.com/jenkinsci/pipeline-stage-view-plugin/pull/182 needs releasing

**Comment by NotMyFault** [comment]
> Looking at open PRs / release status:
> 
> * [Use SVG icons where possible cloudbees-folder-plugin#200](https://github.com/jenkinsci/cloudbees-folder-plugin/pull/200) needs to be merged and released
> * [Let newer Jenkins versions override PNG icons pipeline-stage-view-plugin#182](https://github.com/jenkinsci/pipeline-stage-view-plugin/pull/182) needs releasing

I took a look over all linked PRs and amended them to timja's comment for better accessibility of which one's are still up to do.

To be merged & released:
- https://github.com/jenkinsci/cloudbees-folder-plugin/pull/200
- https://github.com/jenkinsci/implied-labels-plugin/pull/8
- https://github.com/jenkinsci/config-file-provider-plugin/pull/146

To be released:
- https://github.com/jenkinsci/pipeline-stage-view-plugin/pull/182
- https://github.com/jenkinsci/maven-plugin/pull/229
- https://github.com/jenkinsci/jacoco-plugin/pull/172
- https://github.com/jenkinsci/scm-api-plugin/pull/102

Released:
- https://github.com/jenkinsci/docker-plugin/pull/870
- https://github.com/jenkinsci/configuration-as-code-plugin/pull/1802
- https://github.com/jenkinsci/parameterized-trigger-plugin/pull/230
- https://github.com/jenkinsci/nodelabelparameter-plugin/pull/70
- https://github.com/jenkinsci/job-config-history-plugin/pull/132
- https://github.com/jenkinsci/workflow-job-plugin/pull/221
- https://github.com/jenkinsci/workflow-support-plugin/pull/128
- https://github.com/jenkinsci/workflow-cps-plugin/pull/481

**Comment by timja** [comment]
conflicted

**Comment by daniel-beck** [comment]
Per https://issues.jenkins.io/browse/JENKINS-67469 this will break form validation in matrix-auth plugin, which needs to manually reference icons and cannot just use `l:icon`. Is there a solution for this use case that is compatible with both old and new Jenkins releases?

**Comment by timja** [comment]
Removing on-hold. Pretty much everything is released now and it's past the cut-off for the next LTS baseline.

**Review by NotMyFault — APPROVED** [review]
LGTM, my and the other outstanding concerns have been addressed.

The majority of PRs I submitted have been merged and released by now, other ones can be delivered at a later point I guess. Good call to unblock it 👍🏻

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by res0nance — APPROVED** [review]

**Comment by daniel-beck** [comment]
Given that this requires updates of tons of plugins, I'm surprised there are no upgrade notes.

**Comment by timja** [comment]
> Given that this requires updates of tons of plugins, I'm surprised there are no upgrade notes.

I added a somewhat generic update your plugins, not sure how these are phrased normally.

**Comment by NotMyFault** [comment]
> > Given that this requires updates of tons of plugins, I'm surprised there are no upgrade notes.
> 
> I added a somewhat generic update your plugins, not sure how these are phrased normally.

If an upgrade guide (for developers) is needed, it would make sense to outline the impact a little more, I guess?
Could include that static file access via the image path is no longer possible for GIFs and PNGs, that are superseded by SVGs, and it's recommended to access icons via the icon class. That way Jenkins automatically picks a GIF, PNG or SVG, based on the version you're using 🤔 
At least, that was my go-to approach for the PRs.

**Comment by timja** [comment]
The 'upgrade guide' is for administrators not developers, although a developer guide can be added to https://www.jenkins.io/doc/developer/views/

like the https://www.jenkins.io/doc/developer/views/table-to-div-migration/ if there's any volunteers

**Comment by Bananeweizen** [comment]
The removal of "star-gold.png" and "star.png" broke the favorites plugin. See https://github.com/jenkinsci/favorite-plugin/blob/master/src/main/resources/hudson/plugins/favorite/assets.css#L10.

**Comment by daniel-beck** [comment]
@Bananeweizen Please report bugs in the appropriate issue tracker.

**Comment by Bananeweizen** [comment]
@daniel-beck Done. Would it be reasonable to set the PR author as assignee or leave it unassigned to have the whole community grab it?
https://issues.jenkins.io/browse/JENKINS-67737

**Comment by boczh** [comment]
[JENKINS-67176](https://issues.jenkins.io/browse/JENKINS-67176) not only favorite plugin affected when I updated from 2.332 to 2.333... Icons disappeared for the following plugins as well: https://plugins.jenkins.io/job-import-plugin/  and Modules icon disappeared from Maven Project. Solved by reverting back to 2.332

**Comment by MarkEWaite** [comment]
See the ["Report an issue" link](https://www.jenkins.io/participate/report-issue/redirect/#15752) on the [Favorites plugin page](https://plugins.jenkins.io/favorite/) for a guided start to report an issue with the Favorites plugin.

See the ["Report an issue" link](https://www.jenkins.io/participate/report-issue/redirect/#15897) on the [job import plugins page](https://plugins.jenkins.io/job-import-plugin/) for a guided start to report an issue with the job import plugin.

**Comment by basil** [comment]
Causes [JENKINS-68022](https://issues.jenkins.io/browse/JENKINS-68022).

**Comment by ikedam** [comment]
Notes for plugins migrating `${imageURL}` to `<layout:icon>`:

* `<layout:icon>` doesn't accept extra class. Adding a developer-defined class to the class attributes results broken icon: https://issues.jenkins.io/browse/JENKINS-68286
* Icon sizes will be specified with css `width` and `height` in `<layout:icon>`. This means adding `padding` or `border` results smaller-sized icon.
    * Anyway applying `padding` to icons is strange manner and `margin` is better.
    * Also see: https://developer.mozilla.org/en-US/docs/Web/CSS/box-sizing

**Comment by daniel-beck** [comment]
Caused [JENKINS-72673](https://issues.jenkins.io/browse/JENKINS-72673).

**Pull request merged**
